File: 04.Simulation模拟/leetcode_2126_asteroidsDestroyed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    bool asteroidsDestroyed(int mass, vector<int>& asteroids) {
        sort(asteroids.begin(), asteroids.end());
        long long cur = mass;
        for (int i=0; i<asteroids.size(); i++){
            if (cur < asteroids[i]){
                return false;
            }
            cur += asteroids[i];
        }
        return true;
    }
};

int main(){
    Solution s;
    int mass = 10;
    vector<int> asteroids = {3,9,19,5,21};
    cout << s.asteroidsDestroyed(mass, asteroids) << endl;
}

File: 04.Simulation模拟/leetcode_2126_asteroidsDestroyed.py
from typing import List

class Solution:
    def asteroidsDestroyed(self, mass: int, asteroids: List[int]) -> bool:
        asteroids.sort()
        for i in range(len(asteroids)):
            if mass < asteroids[i]:
                return False
            mass += asteroids[i]
        return True

if __name__ == "__main__":
    s = Solution()
    mass = 10
    asteroids = [3,9,19,5,21]
    print(s.asteroidsDestroyed(mass, asteroids))
